### PR TITLE
Create/update model via definition endpoint (fixes #231)

### LINKED ISSUE #231: Add definition PUT, that would create/update models
Would allow JS libs to fetch/save on the same URL

### PR DESCRIPTION
**Pull request by leplatrem**
Allow to GET/PUT definitions on the same API end point

### CODE REVIEW

File: daybed/schemas/relations.py [hunk 1]
```diff
@@ -14,7 +14,7 @@ def __init__(self, db):
 
     def __call__(self, node, value):
         try:
-            self.db.get_model_definition(value)
+            self.db.get_model(value)
```
Comment (Natim) [inline daybed/schemas/relations.py]: I am not understanding why you need to change that and create this method to the backend.

File: daybed/views/models.py [hunk 2]
```diff
@@ -44,6 +46,25 @@ def get_definition(request):
...
+    """Create or update a model definition."""
+    model_id = request.matchdict['model_id']
+    try:
+        model = request.db.get_model(model_id)
+    except ModelNotFound:
+        model = {}
+
+    permissions = invert_permissions_matrix(model.get('permissions', {}))
+    model['permissions'] = permissions
+    model['records'] = []
```
Comment (Natim) [inline daybed/views/models.py]: How does this works with this: https://github.com/spiral-project/daybed/pull/257/files#diff-feded148953d5e4784370906521a1bd2R217 ?

File: daybed/schemas/relations.py (hunk 1 above)
Comment (leplatrem) [inline daybed/schemas/relations.py]: It was just to retrieve definition and permission in one call. Not entirely necessary indeed... I can get rid of it

File: daybed/views/models.py (hunk 2 above)
Comment (leplatrem) [inline daybed/views/models.py]: the `handle_put_model` method will just iterate the empty records collection, but won't erase the existing records

Comment (Natim) [inline daybed/views/models.py]: Ok

**Comment by leplatrem** [comment]
Thanks @Natim, the diff is much better now :)

**Comment by Natim** [comment]
;)

**Pull request merged**